ext/opcache/jit/tls/zend_jit_tls_aarch64.c

```diff
@@ -0,0 +1,247 @@
+		goto code_changed;
+	}
+
+	// ldr x1, [x0, #any]
+	if ((insn[1] & ~AARCH64_LDR_UNSIGNED_IMM_MASK) != 0xf9400001) {
+		zend_accel_error(ACCEL_LOG_DEBUG, "ldr insn does not match: 0x%08" PRIx32 "\n", insn[1]);
+		goto code_changed;
+	}
+
+	// add x0, x0, any
+	if ((insn[2] & ~AARCH64_ADD_IMM_MASK) != 0x91000000) {
+		zend_accel_error(ACCEL_LOG_DEBUG, "add insn does not match: 0x%08" PRIx32 "x\n", insn[2]);
+		goto code_changed;
+	}
+
+	/* Code is intact, we can extract immediate values */
+
+	uint64_t adrp_imm = (uint64_t)( ((insn[0] & 0x00ffffe0) >> 3) | ((insn[0] & 0x60000000) >> 29) ) << 12;
+	uint64_t add_imm = (uint64_t)(insn[2] & AARCH64_ADD_IMM_MASK) >> 10;
+	uint64_t pc = (uint64_t)insn;
+	uintptr_t **where = (uintptr_t**)((pc & ~(4096-1)) + adrp_imm + add_imm);
+
+	/* See https://github.com/ARM-software/abi-aa/blob/2a70c42d62e9c3eb5887fa50b71257f20daca6f9/aaelf64/aaelf64.rst
+	 * section "Relocations for thread-local storage".
+	 * The first entry holds a pointer to the variable's TLS descriptor resolver
+	 * function and the second entry holds a platform-specific offset or
+	 * pointer. */
+	tls_descriptor *tlsdesc = (tls_descriptor*)(where[1]);
+
+	if ((uintptr_t)&_tsrm_ls_cache - (uintptr_t)thread_pointer == (uintptr_t)tlsdesc) {
+		zend_accel_error(ACCEL_LOG_DEBUG, "static tls at offset %p from thread pointer (inferred from tlsdesc)\n", tlsdesc);
+		*tcb_offset = (uintptr_t)tlsdesc;
+		return SUCCESS;
+	}
+
+	*module_index = (tlsdesc->index + DTV_INDEX_GAP) * sizeof(dtv_pointer_t);
+	*module_offset = tlsdesc->offset;
+
+# if ZEND_DEBUG
+	/* We've got the TLS descriptor. Double check: */
+
+	dtv_pointer_t *dtv = *(dtv_pointer_t**)((uintptr_t)thread_pointer + DTV_OFFSET);
+	addr = (void*)(((dtv_pointer_t*)((char*)dtv + *module_index))->val + *module_offset);
+
+	ZEND_ASSERT(addr == &_tsrm_ls_cache);
+# endif
+
+	zend_accel_error(ACCEL_LOG_DEBUG, "dynamic tls module idx %zu offset %zu (inferred from code)\n", (size_t)tlsdesc->index, tlsdesc->offset);
+
+	return SUCCESS;
+
+code_changed:
+
+	/* Code was changed by the linker. Check if we recognize the updated code */
+
+	// movz	x0, #0, lsl #16
+	if ((insn[0] & ~AARCH64_MOVZ_IMM_MASK) != 0xd2a00000) {
+		zend_accel_error(ACCEL_LOG_DEBUG, "movz insn does not match: 0x%08" PRIx32 "\n", insn[0]);
+		return FAILURE;
+	}
+
+	// movk	x0, #0x10
+	if ((insn[1] & ~AARCH64_MOVK_IMM_MASK) != 0xf2800000) {
+		zend_accel_error(ACCEL_LOG_DEBUG, "movk insn does not match: 0x%08" PRIx32 "\n", insn[1]);
+		return FAILURE;
+	}
+
+	// nop
+	for (int i = 0; i < 2; i++) {
+		if (insn[2+i] != AARCH64_NOP) {
+			zend_accel_error(ACCEL_LOG_DEBUG, "nop(%d) insn does not match: 0x%08" PRIx32 "\n", i, insn[2+i]);
+			return FAILURE;
+		}
+	}
+
+	/* Extract immediate values */
+
+	uint64_t movz_imm = (insn[0] & AARCH64_MOVZ_IMM_MASK) >> 5;
+	uint64_t movz_shift = (((insn[0] & AARCH64_MOVZ_HW_MASK) >> 21) << 4);
+	uint64_t mozk_imm = (insn[1] & AARCH64_MOVK_IMM_MASK) >> 5;
```

Comment:
```suggestion
	uint64_t movk_imm = (insn[1] & AARCH64_MOVK_IMM_MASK) >> 5;
```

and also fix it at the uses of this variable names